src/Microsoft.DotNet.Build.CloudTestTasks/WriteBuildStatsJson.cs

```diff
@@ -97,7 +97,8 @@ public TestBuildStatsJson(ITaskItem[] correlationIDs, string logUri, int builtCo
 
         public override bool Execute()
         {
-            using (StreamWriter streamWriter = new StreamWriter(OutputFile))
+            using (var fs = new FileStream(OutputFile, FileMode.OpenOrCreate, FileAccess.Read | FileAccess.Write | FileAccess.ReadWrite))
```

Comment:
Do you really want "OpenOrCreate". It has unexpected behavior if you are writing less data than is currently in the file (it doesn't truncate the existing file, so you'd end up with whatever was left of that file still at the end of yours)
Example: If current file is "aaaaaaa", and you open it and write "bbb", you'd have "bbbaaaa".  I think you want "Create".  Using the "File.Create" helper method should work too